descriptions.md

```diff
@@ -1133,3 +1098,86 @@ Contains fields used when referencing a previous message.
 |---|---|---|---|---|
 | timeserial: String || | REX2a | A unique identifier indicating the message being interacted with. |
 | type: String || | REX2b | The reason for interacting with the previous message, for example, `com.ably.reaction` to indicate a reaction. |
+
+## class `BatchResult<T>`
+
+Contains information about the results of a batch operation.
+
+| Method / Property | Parameter | Returns | Spec | Description |
+|---|---|---|---|---|
+| successCount: Int ||| BAR2a | The number of successful operations |
+| failureCount: Int ||| BAR2b | The number of unsuccessful operations |
+| results: [T] ||| BAR2c | An array of results for the batch operation |
+
+## class `BatchPublishSpec`
+
+Describes the messages that should be published by a batch publish operation, and the channels to which they should be published.
+
+| Method / Property | Parameter | Returns | Spec | Description |
+|---|---|---|---|---|
+| channels: [String] ||| BSP2a | the names of the channels to which all of the messages contained in the `messages` attribute should be published |
+| messages: [Message] ||| BSP2b | the messages which should be published to all of the channels named by the `channels` array |
+
+## class `BatchPublishSuccessResult`
+
+Contains information about the result of successful publishes to a channel requested by a single `BatchPublishSpec`.
+
+| Method / Property | Parameter | Returns | Spec | Description |
+|---|---|---|---|---|
+| channel: String ||| BPR2a | The name of the channel |
+| messageId: String ||| BPR2b | The `messageId` prefix for the published message(s) |
+
+## class `BatchPublishFailureResult`
+
+Contains information about the result of unsuccessful publishes to a channel requested by a single `BatchPublishSpec`.
+
+| Method / Property | Parameter | Returns | Spec | Description |
+|---|---|---|---|---|
+| channel: String ||| BPF2a | The name of the channel |
```

Comment:
```suggestion
| channel: String ||| BPF2a | The name of the channel the message(s) was published to. |
```

Reply:
This would be misleading since in this case the message(s) failed to publish, so i changed it to "the message(s) failed to publish to"

Reply:
Good spot, thanks!